### PR TITLE
New GUI for the native EQ plugin

### PR DESCRIPTION
**Pull request by BaraMGB**
See https://github.com/LMMS/lmms/issues/2347

In my opinion, this is ready. What do you guys think?

### CODE REVIEW

File: plugins/Eq/EqCurve.h [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef EQCURVE_H
+#define EQCURVE_H
+
+#include "qgraphicsitem.h"
+#include "qpainter.h"
+#include "QGraphicsSceneWheelEvent"
```
Comment (tresf) [inline plugins/Eq/EqCurve.h]: I believe this should be formatted:
```cpp
#include <QGraphicsSceneWheelEvent>
```

File: plugins/Eq/.directory [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[Dolphin]
```
Comment (tresf) [inline plugins/Eq/.directory]: The file `plugins/Eq/.directory` should be removed from this commit as it is not needed as part of the LMMS source code.

File: plugins/Eq/EqControlsDialog.cpp [hunk 3]
```diff
@@ -45,113 +45,287 @@ EqControlsDialog::EqControlsDialog( EqControls *controls ) :
 	m_controls = controls;
 	setAutoFillBackground( true );
 	QPalette pal;
-	pal.setBrush( backgroundRole(), PLUGIN_NAME::getIconPixmap( "artwork" ) );
+	pal.setBrush( backgroundRole(), PLUGIN_NAME::getIconPixmap( "EqLayout1BG" ) );
 	setPalette( pal );
-	setFixedSize( 350, 275 );
+	setFixedSize( 500, 500 );
+	QGridLayout * MLayout = new QGridLayout(this);
```
Comment (tresf) [inline plugins/Eq/EqControlsDialog.cpp]: The variable `MLayout ` should follow the same lowercase naming convention used in the rest of the file.

File: plugins/Eq/EqControlsDialog.cpp [hunk 4]
```diff
@@ -45,113 +45,287 @@ EqControlsDialog::EqControlsDialog( EqControls *controls ) :
...
+		m_activeBox->setModel( m_parameterWidget->getBandModels( i )->active );
+
+		// adds the symbols active buttons
+		m_activeBox = new PixmapButton( this, NULL );
+		m_activeBox->setCheckable(true);
+		m_activeBox->setModel( m_parameterWidget->getBandModels( i )->active );
+		switch (i)
+		{
+		case 0:
+			m_activeBox->setActiveGraphic( PLUGIN_NAME::getIconPixmap(
```
Comment (tresf) [inline plugins/Eq/EqControlsDialog.cpp]: Minor formatting, but these 16 `>setActiveGraphic(...)` lines can easily be consolidated into single-line and I'm not sure a line break is warranted before the `break;`

**Comment by tresf** [comment]
This is great.  I haven't reviewed the logic, but I've left a few inline comments in regard to code styling.

Just as an FYI, you no longer need to use spaces to separate parameters from the surrounding parenthesis, although we won't hold up the commit for it, I wanted to let you know... 

So this:

```cpp
m_bandWidthLabel->setText("- " + tr( "Bandwidth")+ " -" );
```

Would previously have been this:
```cpp
m_bandWidthLabel->setText( "- " + tr( "Bandwidth" ) + " -" );
```

Can now be this:
```cpp
m_bandWidthLabel->setText("- " + tr("Bandwidth") + " -");
```

We aren't extremely strict on this specific formatting style yet due to the historical nature of some of these files, but wanted to let you know. :+1:

**Comment by BaraMGB** [comment]
Okay, here we go. @tresf  I think, I'll fixed the criticized things. I'm not sure if "git commit --amend" is here the right way. If not I'll make a clean new PR. Thank you for guiding me. 

the coding convention about the surrounding parenthesis is new to me. The last time I looked into the conventions the spaces was required. Should I go through the code to correct this?

Greetings Steffen

**Comment by tresf** [comment]
> The last time I looked into the conventions the spaces was required. Should I go through the code to correct this?


No, it was more of an FYI.

**Comment by tresf** [comment]
> fixed the criticized things

Very clean, thank you.  Still have to test this.  If someone has time to, please chime ib., especially @curlymorphic as this is the successor to your plugin!

The only thing I noticed this time around is the GAIN algorithm at a glance seems to be a huge, redundant code block that may benefit from being a function, but there's nothing glaringly wrong, so it's fine as-is.

**Comment by BaraMGB** [comment]
Is there any chance to get this in LMMS some day?

**Pull request merged**